src/types/RecordedGame.ts

```diff
@@ -158,4 +160,31 @@ export const MajorGodsByIndex = new Map<number, string>([
 
 export class RecParseError extends Error
 {
+}
+
+
+type MetadataSchemaEntries<T extends (...args: any)=>any> = { [key: string]: {type: T, default: ReturnType<T>, required: boolean} };
+
+function recMetadataSchemaHelper<T extends (...args: any)=>any>(keyArray: ReadonlyArray<string>, valueType: T, defaultValue: ReturnType<T>)
```

Comment:
This is a clever way to build the schemas. I like it. Can you move anything db related to the src/db/mongo/schemas/recordedGame directory? Trying to keep the mongodb schemas and models seperate from the types. 

This specific helper function could move to to the utils.ts. Probably going to  rename /lib to /utils if you want to go ahead and change that before I get to it.